src/traces/sunburst/defaults.js

```diff
@@ -44,7 +44,7 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     if(lineWidth) coerce('marker.line.color', layout.paper_bgcolor);
 
     coerce('marker.colors');
-    var withColorscale = traceOut._hasColorscale = hasColorscale(traceIn, 'marker', 'colors');
+    var withColorscale = traceOut._hasColorscale = (traceIn.marker || {}).coloraxis || hasColorscale(traceIn, 'marker', 'colors');
```

Comment:
This here and in `treemap/defaults.js` would have been nicer as a separate commit.

----

Your new logic is difference than for other colorscale-compatible traces. See for example 

https://github.com/plotly/plotly.js/blob/e778644e1182e13126db6b801af6dd0a386372ba/src/traces/bar/style_defaults.js#L18-L22

- `hasColorscale(traceIn, 'marker', 'colors')` should be enough. For example, we should not coerce the colorscale attributes if `marker.coloraxis` is set, but `marker.colors` isn't a numerical array.

Reply:
cross-referencing https://github.com/plotly/plotly.js/issues/4443#issuecomment-569342008 - this new logic is :ok_hand: for sunburst and treemap traces that support colorscales w/o numerical `marker.colors`

---

Now @archmoj - as this is a special case, could you please add a comment about the

```
 var withColorscale = traceOut._hasColorscale = // ...
```

line saying something like:

```
// N.B. special logic to consider "values" colorscales
```

Reply:
Done in https://github.com/plotly/plotly.js/pull/4444/commits/ef6a95258c8cb83842dbabe7be13a85e1606cf75.

Reply:
Nicely done. Thanks!!